File: sdk/src/main/java/net/m3mobile/sdk/M3Mobile.kt
@file:Suppress("DEPRECATION_ERROR")
package net.m3mobile.sdk

/**
 * Provides access to the M3 SDK.
 * This serves as the main entry point for interacting with the M3 APIs.
 *
 * Use [M3Mobile.instance] to get a concrete implementation of the [M3Sdk] interface.
 */
public object M3Mobile {

    public lateinit var instance: M3Sdk
}

File: dotnet/M3Mobile.M3Sdk.Xamarin/M3Mobile.cs
using System;
using Android.Content;

namespace M3Sdk.Xamarin
{
    /// <summary>
    /// Entry point for creating an M3 Mobile SDK instance.
    /// </summary>
    public static class M3Mobile
    {
        /// <summary>
        /// Creates a native C# SDK instance using the supplied Android context.
        /// Pass <c>Application.Context</c> from a Xamarin.Android application.
        /// </summary>
        /// <param name="context">The Android context used to send broadcasts and read system settings.</param>
        /// <returns>An SDK instance that exposes StartUp, ScanEmul, time, Wi-Fi, and USB APIs.</returns>
        /// <exception cref="ArgumentNullException"><paramref name="context" /> is <c>null</c>.</exception>
        public static IM3Sdk Create(Context context)
        {
            if (context == null)
                throw new ArgumentNullException(nameof(context));

            return new M3Sdk(context);
        }
    }
}
